### PR TITLE
Update deploy file with best practises and example

### PR DESCRIPTION
**Pull request by eothein**
### What are you changing/introducing

Some updates on the deploy

### What is the reason for changing/introducing

Introducing my experience with the deployer

### QA

| Q             | A
| ------------- | ---
| Is bugfix?    | yes/no
| New feature?  | yes/no
| Breaks BC?    | yes/no
| Tests pass?   | yes/no
| Fixed issues  | comma-separated list of tickets # fixed by the PR, if any

### CODE REVIEW

File: docs/guide/concept-depandsync.md [hunk 1]
```diff
@@ -7,21 +7,61 @@ The following is required to reproduce the steps in this guide:
...
+
+### Connection strategies
+
+You can connect to the server either using:
+
+* Pasword `->password('your pwd')`
+* Identity file `->identityFile('~/.ssh/yourkey')`
+* PEM file (e.g. using AWS or Lightsail) `->pemFile('~/.ssh/yourfile.pem')`
+
+We recommend to use either the identity or pem strategie: this way you can commit your `config.php` file on your repo, without having to store a pasword on the repo, which you don't want to do of course.
```
Comment (nadar) [inline docs/guide/concept-depandsync.md]: instead of config.php you mean deploy.php i assume?

File: docs/guide/concept-depandsync.md [hunk 2]
```diff
@@ -7,21 +7,61 @@ The following is required to reproduce the steps in this guide:
...
+
+### Git repository
+
+By using the ssh keys strategy for cloning your project, you don't have to store your github pasword in the config.php file.
```
Comment (nadar) [inline docs/guide/concept-depandsync.md]: same here, i assume you mean deploy.php?

Comment (eothein) [reply docs/guide/concept-depandsync.md]: Indeed sorry

File: docs/guide/concept-depandsync.md (hunk 1 above)
Comment (eothein) [reply docs/guide/concept-depandsync.md]: indeed

**Pull request merged**